### PR TITLE
/order done 

### PR DESCRIPTION
**Pull request by kohchihao**
In this PR, 
- only ongoing orders that have deliverymen assignment can be completed
- the deliveryman should not have the order with him anymore after the command
- the order should still retain the deliveryman (for tracking purposes)
- the order cannot be edited (similar to ongoing)
- unlike pending, completed orders CAN be deleted
- i need help with the test. I am still unsure how to write them.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/order/DoneCommand.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        model.commitOrderBook();
+        model.commitDeliverymenList();
+
+        return new CommandResult(String.format(MESSAGE_COMPLETED_ORDER_SUCCESS, orderToBeCompleted));
+    }
+
+    /**
+     * Remove order from deliveryman.
+     */
+    private static Deliveryman removeOrderFromDeliveryman(Deliveryman targetDeliveryman, Order targetOrder) {
```
Comment (juxd) [inline src/main/java/seedu/address/logic/commands/order/DoneCommand.java]: can we use the same pattern as other `updates` for this? So we create a new `Deliveryman` with the same properties but the order removed, and we use the `ModelManager`'s update foe the deliveryman list. So we can (potentially) allow undoing/redoing

Comment (kohchihao) [reply src/main/java/seedu/address/logic/commands/order/DoneCommand.java]: Since we are not allowing `undo/redo` ill close this.

**Comment by kohchihao** [comment]
@juxd @jinyingtan please review again!

File: src/main/java/seedu/address/logic/commands/order/DoneCommand.java (hunk 1 above)
Comment (juxd) [reply src/main/java/seedu/address/logic/commands/order/DoneCommand.java]: 🆗

File: src/main/java/seedu/address/logic/commands/order/DoneCommand.java [hunk 2]
```diff
@@ -53,12 +54,19 @@ public CommandResult execute(Model model, CommandHistory history) throws Command
         }
 
         orderToBeCompleted.setStatusCompleted();
+
+        //fetch deliveryman from index because order's deliveryman not reliable.
         Deliveryman deliverymanToRemoveOrder = orderToBeCompleted.getDeliveryman();
-        Deliveryman updatedDeliveryman = removeOrderFromDeliveryman(deliverymanToRemoveOrder, orderToBeCompleted);
+        int indexOfDeliveryman = lastShowDeliverymanList.indexOf(deliverymanToRemoveOrder);
+        Deliveryman correctDeliveryman = lastShowDeliverymanList.get(indexOfDeliveryman);
+        Deliveryman updatedDeliveryman = removeOrderFromDeliveryman(correctDeliveryman, orderToBeCompleted);
```
Comment (juxd) [inline src/main/java/seedu/address/logic/commands/order/DoneCommand.java]: will it be cleaner if we do something like 
```
correctDeliveryman = lastShowDeliverymanList.stream()
        .filter(dman -> dman.equals(deliverymanToRemoveOrder))
        .findFirst
        .orElseThrow(…some appropriate exception)
```
This allows for some level of defensiveness as well (ensures that the deliveryman exists, even though it should be the case).
(this is inspired by the `OrderListPanelHandle` in tests.)

File: src/main/java/seedu/address/logic/commands/order/DoneCommand.java [hunk 3]
```diff
@@ -29,6 +29,7 @@
 
     public static final String MESSAGE_COMPLETED_ORDER_SUCCESS = "Order %1$s have been completed.";
     public static final String MESSAGE_ONGOING_ORDER = "Only ONGOING status can be marked as completed.";
+    public static final String MESSAGE_DELIVERYMAN_NOT_EXIST = "Deliveryman does not exist inside deliveryman list.";
```
[the remark this answers is not in the document]
Comment (juxd) [reply src/main/java/seedu/address/logic/commands/order/DoneCommand.java]: ok maybe not this might complicate docs

**Pull request merged**